orc8r/tools/docker/promote.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+set -ex
+
+MAGMA_TAG=1.7
+NEW_MAGMA_TAG=1.7.0
+MAGMA_ARTIFACTORY=artifactory.magmacore.org
+
+declare -A repositories=(
+  [feg]="gateway_go gateway_python"
+  [orc8r]="controller magmalte nginx"
+  [agw]="agw_gateway_c agw_gateway_python ghz_gateway_c ghz_gateway_python"
+  [cwf]="cwag_go gateway_go gateway_pipelined gateway_python gateway_sessiond operator"
+)
+
+for repo in ${!repositories[@]}; do
+  for image in ${repositories[${repo}]}; do
+
+    # Change docker URL to Artifactory
+    sed -i "s/docker/${repo}-prod/g" ~/.docker/config.json
```

Comment:
that can be pulled out above line 17. And is very stateful. Might be better to write temporary configurations and pass `--config=...`.

This also feels like a very unsafe operation, with a lot of unwanted changes

Reply:
You are right @maxhbr, let me look for a fix for this.

Reply:
maybe something like:
```suggestion
    tmpconfig="$(mktemp -d)"
    sed "s/docker/${repo}-prod/g" ~/.docker/config.json > "$tmpconfig/config.json"
```
and then use the ` --config=$tmpconfig...` flag on every docker call.


Also the `sed`, replacing every occurrence of `docker` in the docker config feels way to broad and it should be replaced by a more stable and less error prone pattern. Or maybe even better would be to not use the default config but have it as a heredoc in this script.

Reply:
So we have to deal with 8 docker repos here. I used `sed` command to make the process easy. I understand the tmp file location and everything, but in this script we are not adding config.